### ISSUE TITLE
Feature Request - remove identifier type institutional catalog number

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#7836]
**Is your feature request related to a problem? Please describe.**

A new identifier type is being added, an existing type is clearly being mis-used and causing a great deal of confusion, work, and data which cannot do what it is intended to do. From https://github.com/ArctosDB/arctos/issues/7808#issuecomment-2135899234:

From I am unsure of the status of getting rid of https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type#institutional_catalog_number as part of this. I thought we had an agreement and plan, I'm no longer sure. My comments in https://github.com/ArctosDB/arctos/issues/7808#issuecomment-2128164455 remain valid: there's clearly a great deal of confusion about that type, it is being used for some things that it should not be used for, keeping it around is just going to prolong the pain and make the new thing that much harder to understand. There are 211796 records without valid Arctos record GUIDs using the type at the moment. 128671 of them **do** have an issued-by agent. (And most of those don't seem to be institutions, if anyone needs any more evidence that this is an arbitrary mess.) I would like to

1. for those with an issued-by agent:
    * convert the type to identifier
2. for those without an issued-by agent:
    *  create n 'institutional catalog number' agent, 
    * update all null issuedby to use that agent, then 
    * convert the type to identifier

Note that (2) is in no way a solution, it doesn't magically make these data make sense, it just moves the point of confusion to a hopefully-less-confusing (and more easily documented) place; the identifiers will continue to carry exactly the same amount of information though this process.

Data: https://docs.google.com/spreadsheets/d/1rNlvoWnDrWeVrxD2nWsMAaFO8x3fmBDQ445sWc8DcGQ/edit#gid=1128266536

@mkoo @Jegelewicz @campmlc  please advise and help.

**Describe what you're trying to accomplish** 

Easy access to good data.

**Describe the solution you'd like**

Remove a nonfunctional point of confusion, https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type#institutional_catalog_number

**Describe alternatives you've considered**

Unnecessary struggle resulting in low-quality data.

**Additional context**

https://github.com/ArctosDB/arctos/issues/7808

**Priority**

Nothing has seemed more critical to me in a while...

**Comment by Jegelewicz** [comment]
@dustymc I assume that data is old because I've had you fix some of those? (ALMNH for example)

**Comment by Jegelewicz** [comment]
Can we get an updated list?

**Comment by dustymc** [comment]
Yes the data are stale, I just copied from a comment that was getting buried.

I'll pull fresh data, but possibly not until https://github.com/ArctosDB/arctos/issues/7808 is implemented (it will clear a lot of clutter). 

I will change my priorities if there's any realistic way this can be implemented with https://github.com/ArctosDB/arctos/issues/7808, I believe removing what's clearly a point of confusion at that time would be a HUGE benefit, but my current understanding is that this will not be possible.

**Comment by Jegelewicz** [comment]
@dustymc a couple you can clean up now.

All records in https://arctos.database.museum/search.cfm?guid_prefix=APSU%3AHerp&customoidoper=LIST&oidnum=A-

That have identifiers of type institutional catalog number starting with `A-` should be issued by https://arctos.database.museum/agent/21334600 and can be changed to type identifier

All records in https://arctos.database.museum/search.cfm?guid_prefix=APSU%3AHerp&customoidoper=LIST&oidnum=R-

That have identifiers of type institutional catalog number starting with `R-` should be issued by https://arctos.database.museum/agent/21352850 and can be changed to type identifier

**Comment by dustymc** [comment]
>That have identifiers of type institutional catalog number starting with A- should be issued by https://arctos.database.museum/agent/21334600 and can be changed to type identifier

[temp_APSUHerpA.csv.zip](https://github.com/user-attachments/files/15567599/temp_APSUHerpA.csv.zip)

UPDATE 5713

**Comment by dustymc** [comment]
>That have identifiers of type institutional catalog number starting with R- should be issued by https://arctos.database.museum/agent/21352850 and can be changed to type identifier

[temp_APSUHerpR.csv.zip](https://github.com/user-attachments/files/15567627/temp_APSUHerpR.csv.zip)

UPDATE 2363

**Comment by Jegelewicz** [comment]
@dustymc that is them but missing the issued by.

**Comment by dustymc** [comment]
>missing the issued by

I updated as you asked?!?

**Comment by Jegelewicz** [comment]
@campmlc I have placed a bunch of no-data identifiers in the [identifier unbulkloader](https://arctos.database.museum/loaders/bulkUnLoadIdentifier.cfm). If you could review and let me know if we can just remove them, that would be super helpful! The are under my username and have the following statuses

DGR Bird 
DGR Ento
DGR Mamm
MSB Fish
MSB Herp
MSB Mamm

I think these all got created using a formula in someone's bulkload file. They are either just a bunch of dashes or a letter and some dashes. They don't offer any information and I think it would be good to remove them.

**Comment by campmlc** [comment]
The following marked to autoload in the identifier unbulkloader to remove the legacy data migration institutional catalog number of "M" or "H" etc.
DGR Bird
DGR Ento
DGR Mamm
MSB Fish
MSB Herp
MSB Mamm

**Comment by Jegelewicz** [comment]
@campmlc I do not have access to MSB:Herp. Can you load this to https://arctos.database.museum/loaders/bulkUnLoadIdentifier.cfm?

[cf_temp_unload_identifiers_download(1).csv](https://github.com/user-attachments/files/15592393/cf_temp_unload_identifiers_download.1.csv)

**Comment by dustymc** [comment]
https://docs.google.com/spreadsheets/d/1rNlvoWnDrWeVrxD2nWsMAaFO8x3fmBDQ445sWc8DcGQ/edit#gid=1128266536 is fresh.

 Can we PLEASE immediately proceed as outlined in the initial comment, before this taints the new Arctos record GUID type?

@mkoo help?

**Comment by campmlc** [comment]
I would like at least 24 hours to look over this more, and I'm sure other collections are in the same boat. There are complications here that have not been addressed.

**Comment by Jegelewicz** [comment]
@dustymc all of these ALMNH:Inv can be changed to identifier

[identifierDownload(24).csv](https://github.com/user-attachments/files/15740486/identifierDownload.24.csv)

**Comment by dustymc** [comment]
>all of these ALMNH:Inv can be changed to identifier

done

**Comment by Jegelewicz** [comment]
Actually, I have a lot that can be changed. I just didn't ask because I had assumed they would be changed as part of the switch to Arctos record GUID and I'd rather not spend too much time on it. The sheet won't ever properly open because I guess it is too big? Anyway, here are some that you can change immediately for the entire collection:

NMMNH:Paleo
ALMNH:EH
CRCM:Bird
CRCM:Mamm
UTEP:Herp
APSU:Herp
UNM:MET
ASUMZ:Bivalve

Maybe with those gone the list will be easier to work in.

**Comment by dustymc** [comment]
> assumed they would be changed as part of the switch to Arctos record GUID

So did I but here we are....

>rather not spend too much time on it.

Agree. I can snipe at things if someone wants to deviate from what I laid out [up yonder](https://github.com/ArctosDB/arctos/issues/7836#issue-2333885323), if not then let's just get this gone before someone else finds themselves all tangled up in the arbitrary again, and anything that needs adjusted can be dealt with at any time.

Anyway, whole collections aren't bad (I hope), I'll run those and repull.

**Comment by Jegelewicz** [comment]
> whole collections aren't bad (I hope)

These are all things I either did or had someone do. I can be blamed if anyone yells.

**Comment by dustymc** [comment]
https://github.com/ArctosDB/arctos/issues/7836#issuecomment-2154912123 done, https://docs.google.com/spreadsheets/d/1rNlvoWnDrWeVrxD2nWsMAaFO8x3fmBDQ445sWc8DcGQ/edit#gid=1467013531 updated

**Comment by campmlc** [comment]
Thanks guys, I'll try to look over MSB ones today. @AdrienneRaniszewski  @jldunnum @jtgiermakowski @msbparasites

**Comment by Jegelewicz** [comment]
@dustymc can you update the Google sheet - I'll see if there are any others I can handle.

**Comment by dustymc** [comment]
>update the Google sheet

I did

**Comment by jldunnum** [comment]
Traveling to ASM meeting so can’t assess what is critical here.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by mkoo** [comment]
Please see my email @campmlc

**Comment by Jegelewicz** [comment]
@dustymc these can be changed to identifier

ALMNH"Geo
ALMNH:Inv
ALMNH:Paleo
ASUMZ:Mamm
UTEP:ES

@mkoo it would pare things down if the MVZ collections could be converted as well.

**Comment by Jegelewicz** [comment]
@dustymc 

There are a bunch of these that I think should be type Arctos record GUID based upon the issued by.

guid_prefix | triplet | guid | other_id_type | display_value | id_references | assigned_agent | assigned_date | issued_by_agent | remarks | coll_obj_other_id_num_id
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
CHAS:Bird | CHAS:Bird:7968 | https://arctos.database.museum/guid/CHAS:Bird:7968 | institutional catalog number | 30303 | self | Jessica Weller | 2024-01-12 | Denver Museum of Nature & Science |   | 17754573

**Comment by mkoo** [comment]
> @dustymc these can be changed to identifier
> 
> ALMNH"Geo ALMNH:Inv ALMNH:Paleo ASUMZ:Mamm UTEP:ES
> 
> @mkoo it would pare things down if the MVZ collections could be converted as well.

Sorry-- I thought you knew to do this for MVZ (I'm pretty sure I gave the verbal ok yesterday wtih Carol)- Thanks!

**Comment by Jegelewicz** [comment]
@dustymc all of the CRCM collections can be changed to identifier, but if they have no issued by, it should be https://arctos.database.museum/agent/21346711

**Comment by Jegelewicz** [comment]
@lin-fred the NMMNH:Mamm institutional catalog numbers all appear to be duplicates of the actual catalog number - can they just be removed?

**Comment by lin-fred** [comment]
I don't think I am the one using those numbers? 

See https://github.com/ArctosDB/arctos/issues/6881#event-10825361484

**Comment by Jegelewicz** [comment]
@lin-fred from https://arctos.database.museum/guid/NMMNH:Mamm:4504

![image](https://github.com/ArctosDB/arctos/assets/5725767/0defa8f6-0701-48b7-8d35-4ad3fbadf018)

These are in your records and are redundant?

**Comment by dustymc** [comment]
> type Arctos record GUID based upon the issued by.

I'm probably lost, not sure https://arctos.database.museum/guid/CHAS:Bird:7968 (Passerina) has anything to do with https://arctos.database.museum/guid/DMNS:Bird:30303??

**Comment by Jegelewicz** [comment]
OK fair - I didn't look at the DMNS stuff, so these are who knows.....

**Comment by lin-fred** [comment]
> @lin-fred from https://arctos.database.museum/guid/NMMNH:Mamm:4504
> 
> ![image](https://private-user-images.githubusercontent.com/5725767/337756145-0defa8f6-0701-48b7-8d35-4ad3fbadf018.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3ODU3MjUsIm5iZiI6MTcxNzc4NTQyNSwicGF0aCI6Ii81NzI1NzY3LzMzNzc1NjE0NS0wZGVmYThmNi0wNzAxLTQ4YjctOGQzNS00YWQzZmJhZGYwMTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDdUMTgzNzA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2E2YTRmOTRiMTA0NjcxNTQxY2FlOTc4OGI4MGZjOWFmODdlZjRhM2RhZjE4OTNmNGVhNzIyZWIxNTliN2ViMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.G9DiRSbsaXexQQP4Y8HWoRq3bVnuAnEhA43GMCTx_1U)
> 
> These are in your records and are redundant?

Oh that's weird, I definitely did not add those. I've just looked through our bulkupload of mammals excel sheet and those are not in there. Did they get added because of the reciprocal relationship?

I don't need them there, as you said, its duplication of info.

**Comment by Jegelewicz** [comment]
@campmlc @jldunnum 

from https://arctos.database.museum/guid/MSB:Mamm:287252

I understand that you want to keep these, so I suggest that they be changed to type= identifier and issued by = https://arctos.database.museum/agent/1014941

**Comment by campmlc** [comment]
These were added before the NMMNH migration to Arctos, using Patty Gegick's Excel file for the catalog number notation, when the records were first cataloged at MSB . They are equivalent to the Arctos record guids linking to NMMNHS now. I'll let @jldunnum respond to how he wants them dealt with.

**Comment by Jegelewicz** [comment]
@dustymc in these records - https://arctos.database.museum/search.cfm?guid_prefix=UWBM%3AMamm&customoidoper=LIST&oidtype=institutional%20catalog%20number&oidnum=WDFW%25

the institutional catalog numbers that start with `WDFW` can be changed to identifier and issued by should be added as https://arctos.database.museum/agent/21350366

**Comment by Jegelewicz** [comment]
@dustymc in these records - https://arctos.database.museum/search.cfm?guid_prefix=UWBM%3AMamm&customoidoper=LIST&oidtype=institutional%20catalog%20number&oidnum=MMP%25

the institutional catalog numbers that start with `MMP` can be changed to identifier and issued by should be added as https://arctos.database.museum/agent/1014687

**Comment by Jegelewicz** [comment]
@campmlc I am unloading these. They are all `MSB M -` and like the other batches, it looks like they were added as part of some concatenation for which they had nothing to add to the concatenated value.

[identifierDownload(25).csv](https://github.com/user-attachments/files/15744737/identifierDownload.25.csv)

**Comment by Jegelewicz** [comment]
@dustymc all UWBM:Mamm institutional identifiers can be changed to identifier.

**Comment by Jegelewicz** [comment]
I need fresh data - I keep looking at things I've already cleaned up.

**Comment by campmlc** [comment]
Question as to whether any of this cleanup will affect data in https://arctos.database.museum/info/unreciprocated_relationships.cfm ?

**Comment by dustymc** [comment]
> fresh data

https://docs.google.com/spreadsheets/d/1rNlvoWnDrWeVrxD2nWsMAaFO8x3fmBDQ445sWc8DcGQ/edit#gid=912219362

**Comment by campmlc** [comment]
All MSB:Para records related to institutional catalog numbers issued by the MSB:Host collection agent https://arctos.database.museum/agent/21334672 should all be changed to Arctos record guid relationships.

**Comment by Jegelewicz** [comment]
UTEP:Herb
UTEP:Inv

can be changed to identifier

**Comment by Jegelewicz** [comment]
> All MSB:Para records related to institutional catalog numbers issued by the MSB:Host collection agent https://arctos.database.museum/agent/21334672 should all be changed to Arctos record guid relationships.

@campmlc I think those are all done.

How about all the MSB:Para that have a relationship to an MSB:Mamm identifier? Many of these are odd duplicates. If there is already a link using the same triplet in the GUID, can we get rid of the triplet identifier?

![image](https://github.com/ArctosDB/arctos/assets/5725767/6e90e1b8-35dd-4b09-841e-68cf8b3709f7)

**Comment by Jegelewicz** [comment]
@campmlc and how about these duplications?

https://arctos.database.museum/guid/MSB:Mamm:140646

includes

![image](https://github.com/ArctosDB/arctos/assets/5725767/f0a426fa-7158-4c87-ad82-e7634194c784)

Is that necessary? If so, can we instead say this is an identifier issued by https://arctos.database.museum/agent/21274282

**Comment by Jegelewicz** [comment]
@campmlc see also `SSUC M -` in https://arctos.database.museum/loaders/bulkUnLoadIdentifier.cfm

**Comment by campmlc** [comment]
> > All MSB:Para records related to institutional catalog numbers issued by the MSB:Host collection agent https://arctos.database.museum/agent/21334672 should all be changed to Arctos record guid relationships.
> 
> @campmlc I think those are all done.
> 
> How about all the MSB:Para that have a relationship to an MSB:Mamm identifier? Many of these are odd duplicates. If there is already a link using the same triplet in the GUID, can we get rid of the triplet identifier?
> 
> ![image](https://private-user-images.githubusercontent.com/5725767/337809571-6e90e1b8-35dd-4b09-841e-68cf8b3709f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3OTk2OTMsIm5iZiI6MTcxNzc5OTM5MywicGF0aCI6Ii81NzI1NzY3LzMzNzgwOTU3MS02ZTkwZTFiOC0zNWRkLTRiMDktODQxZS02OGNmOGIzNzA5ZjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDdUMjIyOTUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTgzMTI5Y2I5NmEyNjgyZWNjMzhjOTg1ZjAxN2YxMWYyNzI1YWVkNjI1ZDEzYzM0Njc2ZTQzMWM1Y2U4NmM2NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.pJAspbyHaHq1-9jxUHhdVY28uiE6VuFX69j2dUttlCs)

@jldunnum needs to weigh in on those MSB:Mamm institutional catalog numbers. They were duplicated with the urls for a reason.

**Comment by Jegelewicz** [comment]
@campmlc see also `MSB F -` in https://arctos.database.museum/loaders/bulkUnLoadIdentifier.cfm

**Comment by campmlc** [comment]
> @campmlc see also `SSUC M -` in https://arctos.database.museum/loaders/bulkUnLoadIdentifier.cfm

These should all be institutional catalog number issued by https://arctos.database.museum/agent/21352865. I created the agent and added to the example above.

**Comment by campmlc** [comment]
> > All MSB:Para records related to institutional catalog numbers issued by the MSB:Host collection agent https://arctos.database.museum/agent/21334672 should all be changed to Arctos record guid relationships.
> 
> @campmlc I think those are all done.
> 
> How about all the MSB:Para that have a relationship to an MSB:Mamm identifier? Many of these are odd duplicates. If there is already a link using the same triplet in the GUID, can we get rid of the triplet identifier?
> 
> ![image](https://private-user-images.githubusercontent.com/5725767/337809571-6e90e1b8-35dd-4b09-841e-68cf8b3709f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc4MDA4MTMsIm5iZiI6MTcxNzgwMDUxMywicGF0aCI6Ii81NzI1NzY3LzMzNzgwOTU3MS02ZTkwZTFiOC0zNWRkLTRiMDktODQxZS02OGNmOGIzNzA5ZjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDdUMjI0ODMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzQ3NDkyODNjNDc0MWQ2OWEzMzBmNGJhYjE3NTdiZGVlMTI1NmQyZmEyNTdkOWIxMTRiM2Y3ODU5OTkyNTI1ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.SUiniag6v6XiVegkh6KdIApRW3QqxYhF1ciHPcj9-RA)

MSB:Host records related via institutional catalog numbers to any MSB:Para or MSB:Mamm records should also be converted to Arctos record guids.

**Comment by Jegelewicz** [comment]
> These should all be institutional catalog number issued by https://arctos.database.museum/agent/21352865. I created the agent and added to the example above.

There is no number? Do you want to keep them?

**Comment by Jegelewicz** [comment]
> MSB:Host records related via institutional catalog numbers to any MSB:Para or MSB:Mamm records should also be converted to Arctos record guids.

Those are also done

**Comment by campmlc** [comment]
> > These should all be institutional catalog number issued by https://arctos.database.museum/agent/21352865. I created the agent and added to the example above.
> 
> There is no number? Do you want to keep them?

We need to go through the records and add the issued by collection agent even without the number. Then they can be changed to identifier.

**Comment by campmlc** [comment]
Ditto for all MSB records linked to institutional catalog numbers with prefix "CBF". The issued by agent needs to be added : https://arctos.database.museum/agent/21248995

**Comment by Jegelewicz** [comment]
> We need to go through the records and add the issued by collection agent even without the number. Then they can be changed to identifier.

They are in the issued by bulkload tool. https://arctos.database.museum/loaders/bulk_identifier_issuedby.cfm

**Comment by campmlc** [comment]
Ditto for all MSB Mamm linked to AMNH M numbers: issued by agent needs to be added: https://arctos.database.museum/agent/21347734

**Comment by Jegelewicz** [comment]
> Ditto for all MSB records linked to institutional catalog numbers with prefix "CBF". The issued by agent needs to be added : https://arctos.database.museum/agent/21248995

in the issued by tool - https://arctos.database.museum/loaders/bulk_identifier_issuedby.cfm

**Comment by Jegelewicz** [comment]
> Ditto for all MSB Mamm linked to AMNH M numbers: issued by agent needs to be added: https://arctos.database.museum/agent/21347734

in the issued by loader - https://arctos.database.museum/loaders/bulk_identifier_issuedby.cfm

**Comment by campmlc** [comment]
All MSB linked to DGR anything should have issued by agent as [MSB Division of Genomic Resources.](https://arctos.database.museum/agent/21298561)
Alll MSB:Mamm linked to USNM should have issued by agent https://arctos.database.museum/agent/21347824

I can't do any more of these tonight. Can we get refreshed data with this info added?

**Comment by campmlc** [comment]
Any MSB linked to UNAM should be issued by agent https://arctos.database.museum/agent/21276001

**Comment by Jegelewicz** [comment]
Added to the unloader - https://arctos.database.museum/loaders/bulkUnLoadIdentifier.cfm

MSB B -
None G -

**Comment by Jegelewicz** [comment]
> Ditto for all MSB Mamm linked to AMNH M numbers: issued by agent needs to be added: https://arctos.database.museum/agent/21347734

These were a bit of a mess. Some were formatted as AMNH M xxxxx, some as AMNH xxxxx, some as M-xxxxx. Some had type institutional catalog number, some AMNH: American Museum of Natural History and a few had issued by American Museum of Natural History Mammal Collection. So I downloaded all of these AMNH associated identifiers in all of the formats, determined the unique combinations of Arctos record + AMNH number, and loaded a set of identifiers in the following format:

type | number format | relationship | issued by 
 -- | -- | -- | -- 
identifier | AMNH M xxxxx | same individual as | American Museum of Natural History Mammal Collection 

Then I made a file to unload all of the other combinations of identifier that is now in the [identifier unload tool](https://arctos.database.museum/loaders/bulkUnLoadIdentifier.cfm) with the status `MSB AMNH duplicates`. I did not actually unload these, because I don't feel that is my place. @campmlc @jldunnum please know that this can be undone if for some reason it must be. The file I used to do this work can be found in Dropbox. DGR Conversion Files > Other ID Fixes > AMNH for MSB Mamm.

Going forward, ALL AMNH identifiers should be added as

type | number format | relationship | issued by 
 -- | -- | -- | -- 
identifier | AMNH M xxxxx | same individual as | American Museum of Natural History Mammal Collection 

If we can finish this up and you all can update your data entry workflow, We will be in line with the request at https://github.com/ArctosDB/arctos/issues/5886 which will help us get that issue resolved and everything will be more consistent and easy to find. I'm happy to meet up if we need to.

**Comment by Jegelewicz** [comment]
> Alll MSB:Mamm linked to USNM should have issued by agent https://arctos.database.museum/agent/21347824

These are all done and for a majority of them I was also able to add a link to the USNM records - see https://arctos.database.museum/guid/MSB:Mamm:281784 as an example. Going forward, you all should add this to your workflow when adding USNM mammal identifiers:

## Catalog numbers

Add USNM catalog numbers this way

type | number format | relationship | issued by 
 -- | -- | -- | -- 
identifier | USNM xxxxx | same individual as | Smithsonian Division of Mammals 

You can then search those catalog numbers in the USNM database - https://collections.nmnh.si.edu/search/mammals/
**Hint:** just enter the number in the keyword search field

## Links 

Add links to USNM records this way

type | number format | relationship | issued by 
 -- | -- | -- | -- 
identifier | http://n2t.net/ark:/65665/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx | same individual as | Smithsonian Museum of Natural History EZID 

In this way you can find the "missing links" which are the records with issued by Smithsonian Division of Mammals but no issued by Smithsonian Museum of Natural History EZID OR the records with an identifier that starts with `USNM` but not one that starts with `http://n2t.net/ark:/65665/`.

**Comment by Jegelewicz** [comment]
> All MSB linked to DGR anything should have issued by agent as [MSB Division of Genomic Resources.](https://arctos.database.museum/agent/21298561)

These have all been changed to type = DGR: Division of Genomic Resources (MSB) so that we can deal with them as part of https://github.com/ArctosDB/arctos/issues/5773

**Comment by Jegelewicz** [comment]
> Any MSB linked to UNAM should be issued by agent https://arctos.database.museum/agent/21276001

All of these identifiers were duplicated

![image](https://github.com/ArctosDB/arctos/assets/5725767/02c0a56a-31f1-4ab0-a622-b3fe97886096)

I added issued by to the ones that are formatted UNAM xxxxx

The duplicates (fromatted as issued by UNAM with a bare number) are in the unloader https://arctos.database.museum/loaders/bulkUnLoadIdentifier.cfm
as MSB UNAM dupes @campmlc @jldunnum you can mark to autoload

Also note that I changed the preferred name of the issuing agent. Commas in agent names can cause problems in parsing identifiers. The new preferred name is Universidad Nacional Autonoma de Mexico Museo de Zoologia and the old one is an aka.

Add UNAM catalog numbers this way

type | number format | relationship | issued by 
 -- | -- | -- | -- 
identifier | UNAM xxxxx | same individual as | Universidad Nacional Autonoma de Mexico Museo de Zoologia 

Note however, that all of these have a relationship of `self` and I would think it should be `same individual as` if UNAM is also cataloging them. We can ask Dusty to change them if you would rather have them be `same individual as`

**Comment by Jegelewicz** [comment]
The following would help ensure that everyone is on the same page.

## Enter links to other Arctos records as

identifier type | identifier 
 -- | -- 
Arctos record GUID | https://arctos.database.museum/guid/XXX:Xxxx:12345 

using the right type will mean you don't have to select an issued by (Arctos will do it for you) and you will get a link to the existing record.

## If the related record doesn't exist, enter

identifier type | identifier 
 -- | -- 
identifier | XXX:Xxxx:12345 

So you can find these and update to a link when the related record gets cataloged.

@dustymc does this appear correct? Can we create a tool to help convert the second case when catalog records finally show up?

**Comment by Jegelewicz** [comment]
The MSB:Para relationships to MSB:Host and MSB:Mamm were all over the place. I found them as bare numbers with an issued by, as triplets with no issued by, and as triplets with issued by. 

All of these should now have working links where the related catalog record is in Arctos. BUT there are a bunch of duplicated identifiers like this.

![image](https://github.com/ArctosDB/arctos/assets/5725767/6d209216-82d1-43be-a653-25dc4a081882)

The insitututional catalog number identifiers are redundant and can be removed. They are in the identifier unloader as `MSB:Mamm record GUID dupe` @campmlc @msbparasites  you can mark to autoload to remove them.

**Comment by Jegelewicz** [comment]
After all of the unloading gets done, we can ask for fresh data....

**Comment by dustymc** [comment]
No cleanup is required and none has been requested. No information will be lost when the requested actions are performed. Any needed cleanup was needed before this was opened, and can remain needed after the clearly-confounding factor has been removed. Can we proceed with the initial request?

**Comment by Jegelewicz** [comment]
@dustymc have you created the new agent to be used as issued by?

**Comment by dustymc** [comment]
No

**Comment by Jegelewicz** [comment]
May I?

I was thinking of

type | preferred name | url 
organization | Institutional Catalog | https://github.com/ArctosDB/arctos/issues/7836 

While my first inclination was to use https://arctos.database.museum/agent/0, this provides everyone the ability to find the things that used to be type `institutional catalog number`.

**Comment by dustymc** [comment]
PLEASE do whatever will help get us un-stuck from this place that I can't see why we're stuck in!

>provides everyone the ability to find the things that used to be type institutional catalog number.

Yes, that's what I was trying to say in https://github.com/ArctosDB/arctos/issues/7836#issue-2333885323

(1) (has issuedby) are VERY clearly being used interchangeably, there's nothing to do that I can see.
(2) (no issuedby) are potentially from someone who for some reason hopes this type will convey something, there's an easy way to preserve that. (And hopefully someone will do something better with those poor data, but that doesn't need to stop progress either.)

Do please be verbose in the remarks-or-something - "legacy mess-marker, don't use this for any purpose" or something of the sort - and hopefully we'll be able to bad dup it before anyone does anything irrecoverable....

**Comment by mkoo** [comment]
While I like the poetic use of "Agent Zero" I think we do need a separate one as you suggest for Institutional Catalog.

If you create it, we can update but a profile of "This is used for legacy records marked with "institutional catalog number". Currently all assignable values are attributed to actual institutions and identifier values. This agent should not be used in the future but is simply a legacy notation" or similar would be helpful

**Comment by Jegelewicz** [comment]
> hopefully we'll be able to bad dup it before anyone does anything irrecoverable....

Can't we do that immediately after the transition? It would still show up on the transitioned identifiers, but nobody could use it going forward?

**Comment by campmlc** [comment]
There are still records in the bulkloader with institutional catalog number.

**Comment by Jegelewicz** [comment]
https://arctos.database.museum/agent/21352872

**Comment by Jegelewicz** [comment]
> There are still records in the bulkloader with institutional catalog number.

Those can be transitioned as part of this process.

**Comment by campmlc** [comment]
> > There are still records in the bulkloader with institutional catalog number.
> 
> Those can be transitioned as part of this process.

Will this happen automatically or is that the responsibility of the associated collection?

**Comment by Jegelewicz** [comment]
@dustymc see above.

**Comment by dustymc** [comment]
I can update bulkloader with the same path.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
What is happening here? We use institutional catalog number.

**Comment by mkoo** [comment]
There's now a good agent that works as a proxy for Institutional Catalog (https://arctos.database.museum/agent/21352872) for when we dont know the institution. Thanks @Jegelewicz for making it. 

This will be the Issuedby for all the null Issued by where identifier type is the now deprecated institutional catalog number. The only way that value has value is if there's an explicit Issuedby, and for as many as we can these have been added or cleaned up by all collections. This allows us to do much more automated integration of records. We have a new type ```Arctos record GUID``` that can do lots of remarkable things now for Arctos collection triples. If you missed the relationship_bot demo, we have a recording from the AWG meeting.

@dustymc can you confirm that stuff in the bulkloader can be updated by you? or should those be redone by collections (How many records are there @campmlc ?)

**Comment by campmlc** [comment]
> There's now a good agent that works as a proxy for Institutional Catalog (https://arctos.database.museum/agent/21352872) for when we dont know the institution. Thanks @Jegelewicz for making it.
> 
> This will be the Issuedby for all the null Issued by where identifier type is the now deprecated institutional catalog number. The only way that value has value is if there's an explicit Issuedby, and for as many as we can these have been added or cleaned up by all collections. This allows us to do much more automated integration of records. We have a new type `Arctos record GUID` that can do lots of remarkable things now for Arctos collection triples. If you missed the relationship_bot demo, we have a recording from the AWG meeting.
> 
> @dustymc can you confirm that stuff in the bulkloader can be updated by you? or should those be redone by collections (How many records are there @campmlc ?)

I manually updated the current ones I have access to in MSB:Mamm, but there is continuous data entry going on across divisions at MSB and I imagine there will be others.

**Comment by dustymc** [comment]
>bulkloader can be updated by you

yes https://github.com/ArctosDB/arctos/issues/7836#issuecomment-2159399059

**Comment by campmlc** [comment]
It would also be super helpful if there were a means in place, or a plan for a means, to deal with temporal lags between cataloging of related items, both current records and future entries, before this goes into effect.

**Comment by campmlc** [comment]
> > bulkloader can be updated by you
> 
> yes [#7836 (comment)](https://github.com/ArctosDB/arctos/issues/7836#issuecomment-2159399059)

Any possibility of getting the Browse and Edit SQL tool functional? Currently not possible for operators to make bulk edits to the bulkloader files. I had to make the edits I performed one record at a time in Browse and Edit. The alternative, downloading, deleting, and rebulkloading, is an enormous amount of work (Excel date formats, for one, and hundreds of empty columns to deal with for another), and provides an open path to introduce error.

**Comment by mkoo** [comment]
How about a **_banner_** display?-- we have had several lags in implementation since this has been on the docket of agendas for a while and we have had a week since we made decisions. I agree though we need to push notifications to people. 

I will post a banner to read (suggestions welcomed):
Please do not use identifier type  of 'institutional catalog number' as we are creating better tools for Identifiers. Instead ensure that IssuedBy has the proper Institutional Agent (or make one). Please see the updated Handbook: https://handbook.arctosdb.org/how_to/How-to-choose-the-proper-other-identifier-type.html

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
Issued by is not a substitute for saying that something is a catalog number? All of our institutional catalog numbers should have issuers.

**Comment by Jegelewicz** [comment]
> It would also be super helpful if there were a means in place, or a plan for a means, to deal with temporal lags between cataloging of related items, both current records and future entries, before this goes into effect.

https://github.com/ArctosDB/arctos/issues/7836#issuecomment-2156208125

**Comment by campmlc** [comment]
> > It would also be super helpful if there were a means in place, or a plan for a means, to deal with temporal lags between cataloging of related items, both current records and future entries, before this goes into effect.
> 
> [#7836 (comment)](https://github.com/ArctosDB/arctos/issues/7836#issuecomment-2156208125)

The only problem with that is during bulkload, operators from one collection do not necessarily know at the time of data entry whether or not the record exists at another collection in order to alter data entry type and workflows. And there should be some automation that would allow these to be converted as matches become available.

**Comment by campmlc** [comment]
When we had the "MSB:Mamm" and "MSB:Para" ID types, we could choose one of those to create the relationship, and if the corresponding record was not yet cataloged, the link would 404, which wasn't great, but at least the relationship was captured. And those could be found and fixed, even by the bot. If we get rid of institutional catalog number and force people to hand-type some version of the catalog number as an "identifier" - how do we prevent loss of data?

**Comment by jldunnum** [comment]
When all identifiers are ending up in a single field it makes managing them very difficult. The ability to select specific independant identifiers for display and sorting in their own column is very helpful when managing our collection. We can still do this for our NK number but having that for others is still needed. 
For example this morning our funders at USGS just requested a download of the 51K USGS specimen records we hold at the MSB with their Institutional Catalog number. I could not just give them a list that they could easily sort. The United States Geological Survey identifier is currently sitting in with all the other identifiers. Also no option to dispay "issued by" on the customize search & results either. 
We have many records that contain multiple identifiers, including multiple institutional catalog numbers that we want to be able to parse out.  
For example: MSB:Mamm:124975	collector number EWV 1365 | United States Geological Survey identifier BS/FC 24975 | NCBI BioSample identifier https://www.ncbi.nlm.nih.gov/biosample/SAMN23585490 | NK 34841 | U. S. National Park Service catalog SAPU 45036

**Comment by dustymc** [comment]
> very difficult


----

<img width="374" alt="Screenshot 2024-06-24 at 11 04 37" src="https://github.com/ArctosDB/arctos/assets/5720791/3337ca4e-1aed-4945-87f3-080aa8b06f9a">


----

<img width="264" alt="Screenshot 2024-06-24 at 11 04 57" src="https://github.com/ArctosDB/arctos/assets/5720791/7051c99c-bc49-490b-ac0e-94f66f811279">

----

<img width="729" alt="Screenshot 2024-06-24 at 11 05 14" src="https://github.com/ArctosDB/arctos/assets/5720791/c66b6134-2211-4549-9517-820383b4a4d2">

----
<img width="807" alt="Screenshot 2024-06-24 at 11 19 11" src="https://github.com/ArctosDB/arctos/assets/5720791/daf3249f-0765-4f91-8a80-6cbdbea3a34d">

**Comment by campmlc** [comment]
I think what @Jon is requesting is the option to parse out selected identifiers and issued by in a separate column in the main search results, as below. I don't think anyone can expect that USGS would be able to figure out how to use our identifiers download with merge option - especially given that attempting this with the 51000+ USGS records and over 100,000 rows of identifiers at MSB results in browser collapse. 

![image](https://github.com/ArctosDB/arctos/assets/14808196/a4704a7e-3241-4254-a351-a9a2c0c014b7)

**Comment by jldunnum** [comment]
Thanks for that route to a download Dusty and yes as @campmlc that would be a very useful way to still parse identifiers out and sort/search for various subsets or premutations of the data (e.g., looking at all the USGS holdings but then checking which are also NPS specimens)

**Comment by dustymc** [comment]
>main search results

The identifier-based model is no obstacle to that, there's a fairly elaborate demo Issue (https://github.com/ArctosDB/arctos/issues/6983). Adding a specific type would be a cost, but it's the same cost as any column. What is a barrier is having lots of ways to do things, and we do because we can't move on issues like this one.

>browser collapse.

Part of that is related to the problem that I'm trying to solve in this Issue; it takes orders of magnitude more processors than it should do do anything, and that still results in bad data because we're demonstrably entering arbitrary things, which is inevitable as long as we have a sea of options. (See also https://github.com/ArctosDB/arctos/issues/7837)

**Comment by ebraker** [comment]
@ArctosDB/arctos-code-table-administrators need a more robust definition (e.g., bonafide catalog number)

**Comment by campmlc** [comment]
Some resources for coming up with a more robust definition:

[Guidance for Cataloging DOI Museum Collections](https://www.doi.gov/sites/doi.gov/files/uploads/museum_cataloging_guidance_march_2016_fnl.pdf)

https://www.nps.gov/museum/publications/MHII/mushbkII.html

https://americanornithology.org/professional-resources/collections-permitting-resources/guidelines-for-citing-museum-collections/

http://collections.paleo.amnh.org/11
chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/http://collections.paleo.amnh.org/assets/AMNH_Cataloguing_Procedures.pdf
